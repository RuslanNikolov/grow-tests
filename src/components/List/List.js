import React from "react";
import ListItem from "../ListItem/ListItem";

const List = props => {
  const { items } = props;

  return items.length > 0 ? (
    items.map(item => <ListItem key={item.id} {...item} />)
  ) : (
      <div className="no-items"> No items found.</div>
    );
};
export default List;
