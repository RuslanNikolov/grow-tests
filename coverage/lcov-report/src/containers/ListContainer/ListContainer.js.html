<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/ListContainer/ListContainer.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/containers/ListContainer</a> ListContainer.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/15</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { useEffect, useState } from "react";
import List from "../../components/List/List";
import { getData } from "../../helpers/requests";
import { transformData } from "../../helpers/transform";
&nbsp;
async function loadData(setError, setItems, setIsLoading) {
  const data = await getData();
<span class="cstat-no" title="statement not covered" >  setError(data.error);</span>
<span class="cstat-no" title="statement not covered" >  setItems(transformData(data.items));</span>
<span class="cstat-no" title="statement not covered" >  setIsLoading(false);</span>
}
&nbsp;
const ListContainer = props =&gt; {
  const [items, setItems] = useState([]);
  const [error, setError] = useState("");
  const [isLoading, setIsLoading] = useState(false);
&nbsp;
  useEffect(() =&gt; {
    loadData(setError, setItems, setIsLoading);
  }, []);
&nbsp;
  const onClick = <span class="fstat-no" title="function not covered" >as</span>ync e =&gt; {
<span class="cstat-no" title="statement not covered" >    setIsLoading(true);</span>
<span class="cstat-no" title="statement not covered" >    props.onClick &amp;&amp; props.onClick();</span>
<span class="cstat-no" title="statement not covered" >    loadData(setError, setItems, setIsLoading);</span>
  };
&nbsp;
  return (
    &lt;&gt;
      &lt;input
        type="button"
        value="Refresh"
        className="button"
        onClick={onClick}
        disabled={isLoading}
      /&gt;
      {error ? <span class="branch-0 cbranch-no" title="branch not covered" >&lt;div className="error"&gt;{error}&lt;/div&gt; </span>: &lt;List items={items} /&gt;}
    &lt;/&gt;
  );
};
&nbsp;
export default ListContainer;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 15 2019 21:43:50 GMT+0300 (Eastern European Summer Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
