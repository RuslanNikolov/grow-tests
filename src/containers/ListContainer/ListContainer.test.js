import * as React from 'react'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ListContainer from './ListContainer'
import List from '../../components/List/List'
import * as ListContainerExports from './ListContainer'

jest.mock('../../helpers/requests');


describe('ListContainer', () => {
    let wrapper;
    let loadDataSpy;
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation(init => [init, setState]);

    beforeEach(() => {
        loadDataSpy = jest.spyOn(ListContainerExports, 'loadData')
        wrapper = mount(<ListContainer />)
    })

    afterEach(() => {
        jest.clearAllMocks()
        wrapper.unmount()
    })

    it('should render the correct initial UI', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('should have correct props', () => {
        expect(wrapper.props()).toEqual({})
    })

    it('should have initially no items', () => {
        const ListComponent = wrapper.find(List)
        expect(ListComponent.prop('items')).toEqual([]);
    })

    it('should have initially no error', () => {
        const errorDiv = wrapper.find('.error')
        expect(errorDiv).not.toExist()
    })

    it('should initially have no loading', () => {
        const button = wrapper.find('.button');
        expect(button.prop('disabled')).toBeFalsy()
    })

    it('should load data on initial load', () => {
        expect(loadDataSpy).toHaveBeenCalled();
    })

    it('should call useState', () => {
        wrapper.find('.button').props().onClick()
        expect(setState).toHaveBeenCalled();
    })

    it('should reload data on button click', () => {
        wrapper.find('.button').props().onClick()
        expect(loadDataSpy).toHaveBeenCalled()
    })
    

})
