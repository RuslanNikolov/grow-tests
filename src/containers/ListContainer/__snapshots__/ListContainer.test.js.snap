// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer should render the correct initial UI 1`] = `
<ListContainer>
  <input
    className="button"
    disabled={false}
    onClick={[Function]}
    type="button"
    value="Refresh"
  />
  <List
    data-testid="list"
    items={Array []}
  >
    <div
      className="no-items"
    >
       No items found.
    </div>
  </List>
</ListContainer>
`;
